src/inabox/inabox-viewport.js

```diff
@@ -92,6 +98,314 @@ export function resetBodyForOverlay(win, bodyElement) {
   });
 }
 
+/**
+ * This object represents the viewport. It tracks scroll position, resize
+ * and other events and notifies interesting parties when viewport has changed
+ * and how.
+ * @implements {ViewportInterface}
+ */
+class InaboxViewportImpl {
+  /**
+   * @param {!../service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!ViewportBindingDef} binding
+   */
+  constructor(ampdoc, binding) {
+    const {win} = ampdoc;
+
+    /** @const {!../service/ampdoc-impl.AmpDoc} */
+    this.ampdoc = ampdoc;
+
+    /** @const {!ViewportBindingDef} */
+    this.binding_ = binding;
+
+    /**
+     * Used to cache the rect of the viewport.
+     * @private {?../layout-rect.LayoutRectDef}
+     */
+    this.rect_ = null;
+
+    /** @private @const {!Observable<!../service/viewport/viewport-interface.ViewportChangedEventDef>} */
+    this.changeObservable_ = new Observable();
+
+    /** @private @const {!Observable} */
+    this.scrollObservable_ = new Observable();
+
+    /** @private @const {!Observable<!../service/viewport/viewport-interface.ViewportResizedEventDef>} */
+    this.resizeObservable_ = new Observable();
+
+    /** @private @const {boolean} */
+    this.useLayers_ = isExperimentOn(win, 'layers');
+    if (this.useLayers_) {
+      installLayersServiceForDoc(
+        ampdoc,
+        this.binding_.getScrollingElement(),
+        true
+      );
+    }
+
+    this.binding_.connect();
+    this.binding_.onScroll(this.scroll_.bind(this));
+    this.binding_.onResize(this.resize_.bind(this));
+
+    // Top-level mode classes.
+    const docElement = win.document.documentElement;
+    docElement.classList.add('i-amphtml-singledoc');
+    docElement.classList.add('i-amphtml-standalone');
+    if (isIframed(win)) {
+      docElement.classList.add('i-amphtml-iframed');
+    }
+  }
+
+  /** @override */
+  dispose() {
+    this.binding_.disconnect();
+  }
+
+  /** @override */
+  ensureReadyForElements() {}
+
+  /** @override */
+  getPaddingTop() {
+    return 0;
+  }
+
+  /** @override */
+  getScrollTop() {
+    return this.binding_.getScrollTop();
```

Comment:
Is this not always 0? Or is this still meant to be global scroll position?

Reply:
this is global